geopandas/geoseries.py

```diff
@@ -170,7 +168,10 @@ def _wrapped_pandas_method(self, mtd, *args, **kwargs):
 
     def __getitem__(self, key):
         if isinstance(key, (slice, list, Series, np.ndarray)):
-            return super(GeoSeries, self).__getitem__(key)
+            block = self._data._block[key]
+            index = self.index[key]
+            return GeoSeries(SingleBlockManager(block, axis=index),
```

Comment:
Hmm, `test_align` was already passing I thought. 

(BTW, that's the reason there are still those messy print statements, to be able to follow what is happening and eg to see when densification occurs)